### ISSUE TITLE
change in behavior of drawKImage

### ISSUE THREAD

**Issue by esheldon** [GalSim-developers/GalSim#797]
I'm noticing a difference in the version of the code I had been using

993044f1eb13d577e27d6f48fed23b5059ed8387

and recent galsim 1.4.2

In this code, the new version consistently creates a k image of size `[44,44]` while the old code gives a large variation of sizes.

Is this expected?

```python
import numpy
import galsim

sigma=1.414
scale=1.0

g=galsim.Gaussian(sigma=sigma)

dim=32
im = g.drawImage(nx=dim,ny=dim,scale=scale)

error=0.001
im.array[:,:] += numpy.random.normal(
    size=im.array.shape,
    scale=error,
)


iim = galsim.InterpolatedImage(im, scale=scale)

kscale=0.15
kim,kim_imag = iim.drawKImage(scale=kscale)
print("k shape:",kim.array.shape)
```

**Comment by rmjarvis** [comment]
If the commit where the behavior changes is 156597f2, then yes this is probably expected.  At least that commit involved a bug fix related to drawKImage when setting a scale value.  Does the new behavior seem to be wrong to you in some way?

**Comment by esheldon** [comment]
Other than this difference, I haven't yet seen any smoking gun that one is better than the other.  I'll be doing more tests.

**Comment by rmjarvis** [comment]
Closing this on the presumption that the old version was wrong, and bug fix in 156597f is responsible for the change in behavior.  Please reopen if you think the new behavior is incorrect in some way.
